include/nbl/builtin/hlsl/bxdf/brdf/specular/common.hlsl

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_BXDF_BRDF_SPECULAR_COMMON_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BXDF_BRDF_SPECULAR_COMMON_INCLUDED_
+
+#include <nbl/builtin/hlsl/bxdf/common.hlsl>
+#include <nbl/builtin/hlsl/bxdf/ndf/common.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace bxdf
+{
+namespace brdf
+{
+namespace specular
+{
+
+template <class fresnel_t, class ndf_t, class Sample, class Interaction, class MicrofacetCache>
+struct CookTorrance : BxDFBase<float3, float, Sample, Interaction, MicrofacetCache>
```

Comment:
why are we making `Sample`, `Interaction` and `MicrofacetCache` template parameters, I think only the RayDirDistribution ought to be one (which then goes into the interaction and sample)

Reply:
but u dont know if u want isotropic or anisotropic interaction at this point

Reply:
btw could we make the `MicrofacetCache`'s have a `using` typedef for the `surface_interaction` (iso vs aniso) it wants?